docs/tutorial/fastapi.rst

```diff
@@ -0,0 +1,421 @@
+.. _build-a-rock-for-a-fastapi-application:
+
+Build a rock for a FastAPI application
+--------------------------------------
+
+In this tutorial, we'll create a simple FastAPI application and learn how to
+containerise it in a rock, using Rockcraft's ``fastapi-framework``
+:ref:`extension <fastapi-framework-reference>`.
+
+Setup
+=====
+
+.. include:: /reuse/tutorial/setup.rst
+
+.. note::
+    This tutorial requires the ``latest/edge`` channel of Rockcraft. Use
+    ``sudo snap refresh rockcraft --channel latest/edge`` to get the latest
+    edge version.
+
+Finally, create a new directory for this tutorial and go inside it:
+
+.. code-block:: bash
+
+   mkdir fastapi-hello-world
+   cd fastapi-hello-world
+
+Create the FastAPI application
+==============================
+
+Start by creating the "Hello, world" FastAPI application that will be used for
+this tutorial.
+
+Create a ``requirements.txt`` file, copy the following text into it and then
+save it:
+
+.. literalinclude:: code/fastapi/requirements.txt
+
+In order to test the FastAPI application locally (before packing it into a
+rock), install ``python3-venv`` and create a virtual environment:
+
+.. literalinclude:: code/fastapi/task.yaml
+    :language: bash
+    :start-after: [docs:create-venv]
+    :end-before: [docs:create-venv-end]
+    :dedent: 2
+
+In the same directory, copy and save the following into a text file called
+``app.py``:
+
+.. literalinclude:: code/fastapi/app.py
+    :language: python
+
+Run the FastAPI application using ``fastapi dev app.py --port 8000`` to verify
+that it works.
+
+Test the FastAPI application by using ``curl`` to send a request to the root
+endpoint. You may need a new terminal for this, if you are using Multipass use
+``multipass shell rock-dev`` to get another terminal:
+
+.. literalinclude:: code/fastapi/task.yaml
+    :language: bash
+    :start-after: [docs:curl-fastapi]
+    :end-before: [docs:curl-fastapi-end]
+    :dedent: 2
+
+The FastAPI application should respond with ``{"message":"Hello World"}``.
+
+The Django application looks good, so let's stop it for now by pressing
```

Comment:
```suggestion
The application looks good, so let's stop it for now by pressing
```

"Django" is for sure wrong but we just said "The FastAPI application" on the previous line, so I think this is enough

Reply:
fixed